eessi/testsuite/tests/apps/osu.py

```diff
@@ -0,0 +1,309 @@
+"""
+This module tests the binary 'osu' in available modules containing substring 'OSU-Micro-Benchmarks'.
+The basic application class is taken from the hpctestlib to which extra features are added.
+"""
+import reframe as rfm
+from hpctestlib.microbenchmarks.mpi.osu import osu_benchmark
+
+from eessi.testsuite import hooks, utils
+from eessi.testsuite.constants import *
+from eessi.testsuite.utils import find_modules, log
+
+
+def my_filtering_function():
+    """
+    Filtering function for filtering scales for the pt2pt OSU test
+    """
+    return [
+        k for (k, v) in SCALES.items()
+        if v['num_nodes'] * v.get('num_cpus_per_node', 0) == 2
+        or (v['num_nodes'] == 2 and v.get('node_part', 0) == 1)
+        or (v['num_nodes'] == 1 and v.get('node_part', 0) == 1)
+    ]
+
+
+def my_filtering_function_coll():
+    """
+    Filtering function for filtering scales for collective the OSU test
+    """
+    return [
+        k for (k, v) in SCALES.items()
+        if (v['num_nodes'] * v.get('num_cpus_per_node', 1) > 1)
+        or (v.get('node_part', 0) > 0)
+    ]
+
+
+@rfm.simple_test
+class osu_pt_2_pt(osu_benchmark):
+    ''' Run-only OSU test '''
+    scale = parameter(my_filtering_function())
+    valid_prog_environs = ['default']
+    valid_systems = []
+    time_limit = '30m'
+    module_name = parameter(find_modules('OSU-Micro-Benchmarks'))
+    # Device type for non-cuda OSU-Micro-Benchmarks should run on hosts of both
+    # node types. To do this the default device type is set to GPU.
+    device_type = parameter([DEVICE_TYPES[CPU], DEVICE_TYPES[GPU]])
```

Comment:
we should here unset `num_tasks_per_node` (as set by the hpctestlib), so we can set it ourselves in the hook:
```suggestion
    device_type = parameter([DEVICE_TYPES[CPU], DEVICE_TYPES[GPU]])
    num_tasks_per_node = None
```

Reply:
Incorporated.